src/recovery/mod.rs

```diff
@@ -855,22 +861,18 @@ impl Recovery {
     }
 
     #[cfg(feature = "qlog")]
-    pub fn to_qlog(&self) -> qlog::event::Event {
-        // QVis can't use all these fields and they can be large.
-        qlog::event::Event::metrics_updated(
-            Some(self.min_rtt.as_secs_f32() * 1000.0),
-            Some(self.rtt().as_secs_f32() * 1000.0),
-            Some(self.latest_rtt.as_secs_f32() * 1000.0),
-            Some(self.rttvar.as_secs_f32() * 1000.0),
-            None, // delay
-            None, // probe_count
-            Some(self.cwnd() as u64),
-            Some(self.bytes_in_flight as u64),
-            Some(self.ssthresh as u64),
-            None, // packets_in_flight
-            None, // in_recovery
-            None, // pacing_rate
-        )
+    pub fn maybe_qlog(&mut self) -> Option<qlog::event::Event> {
+        let qlog_metrics = QlogMetrics {
+            min_rtt: self.min_rtt,
+            smoothed_rtt: self.rtt(),
+            latest_rtt: self.latest_rtt,
+            rttvar: self.rttvar,
+            cwnd: self.cwnd() as u64,
+            bytes_in_flight: self.bytes_in_flight as u64,
+            ssthresh: self.ssthresh as u64,
+        };
+
+        self.qlog_metrics.maybe_update(qlog_metrics)
```

Comment:
Why not do something like:
```rust
if self.qlog_metrics != qlog_metrics {
    self.qlog_metrics = qlog_metrics;

    return Some(qlog::event::Event::metrics_updated(
        ...
    ));
}

None
```
instead of comparing each field manually?

Reply:
We only want create an event with the changed fields in it. The common case is that bytes in flight changes often while the other field don't, so comparing the objects will probably return true and then we'd have to go through each one to figure out which to put in the new event that gets emitted.

Reply:
Oh, I missed that this would create partial events as well. There's probably a way to clean this up a bit and make it less repetitive, but we can look at it in a future change.